app/components/display/search_result_component/search_result_component.html.haml

```diff
@@ -1,20 +1,27 @@
-.search-result-component{class: sub_component_class, 'data-controller': 'reveal-component'}
-    %a.title{href: @link}
-        = @title
+.search-result-component{class: sub_component_class, 'data-controller': 'reveal-component federation', 'data-federation-config-value': federated_portals.to_h.to_json}
+    %a.title{href: @link, style: @portal_color ? "color: #{@portal_color} !important" : '', target: @portal_color ? "_blank" : ''}
+        .d-flex.align-items-center
+            = @title
+            = inline_svg_tag 'icons/external-link.svg', class: "ml-1 federated-icon-#{@portal_name} #{@portal_color ? '' : 'd-none'}"
+
     - if @uri
         .uri
             = @uri
     - if @definition
-
         = display_in_multiple_languages(@definition)
     .actions
-        = details_button
-        = visualize_button
-        = mappings_button
+        - unless @portal_name 
+            = details_button
+            = visualize_button
+            = mappings_button
         - if subresults?
             = reveal_ontologies_button("#{subresults.size} #{t('search.result_component.more_from_ontology')}", sub_ontologies_id, 'icons/three-dots.svg')
         - if reuses?
             = reveal_ontologies_button("#{t('search.result_component.reuses_in')} #{reuses.size} ontologies", reuses_id, 'icons/reuses.svg')
+        - if @portal_name
```

Comment:
here why have two blocks for the same thing?

Reply:
Because the first one is for the regular results, and the second one is for the subresults

Reply:
It needs to be harmonized in both cases, you show a button, whatever regular or sub result.  
A possible solution is `@other_portals << {name: @portal_name, color: @portal_color, light_color: @portal_light_color, link: @link}` and remove line 21 and 22.

Reply:
done